File: examples/fib.js
// % node --experimental-options --engine.CompileOnly=fib --engine.Inlining=false --vm.Dgraal.Dump=Truffle:2 -f fib.js 14

function fib(n) {
  if (n <= 1) {
    return n;
  } else {
    return fib(n - 1) + fib(n - 2);
  }
}

while (true) {
  for (let n = 2; n < process.argv.length; n++) {
    fib(parseInt(process.argv[n]));
  }
}

File: examples/fib.rb
# % ruby --jvm --experimental-options --engine.CompileOnly=fib --engine.TraceCompilation --engine.NodeSourcePositions --engine.MultiTier=false --engine.Inlining=false --vm.Dgraal.Dump=Truffle:1 --vm.Dgraal.PrintBackendCFG=true fib.rb 14

def fib(n)
  if n <= 1
    n
  else
    fib(n - 1) + fib(n - 2)
  end
end

loop do
  ARGV.each do |arg|
    fib(Integer(arg))
  end
end
